qcodes/instrument_drivers/AlazarTech/ATS.py

```diff
@@ -935,54 +994,117 @@ class Buffer:
         number_of_channels: the number of channels that will be stored in the
             buffer
     """
+    logger = False
+
     def __init__(self, bits_per_sample, samples_per_buffer,
                  number_of_channels):
+
         if os.name != 'nt':
             raise Exception("Buffer: only Windows supported at this moment")
+
+        self.samples_per_buffer = samples_per_buffer
+        self.number_of_channels = number_of_channels
+
         self._allocated = True
 
-        bytes_per_sample = int((bits_per_sample + 7)//8)
-        np_sample_type = {1: np.uint8,
-                          2: np.uint16}[bytes_per_sample]
+        self.bytes_per_sample = int((bits_per_sample + 7)//8)
+        self.np_sample_type = {1: np.uint8, 2: np.uint16}[self.bytes_per_sample]
 
         # try to allocate memory
         mem_commit = 0x1000
         page_readwrite = 0x4
 
-        self.size_bytes = bytes_per_sample * samples_per_buffer * \
+        self.size_bytes = self.bytes_per_sample * samples_per_buffer * \
                           number_of_channels
 
         # for documentation please see:
         # https://msdn.microsoft.com/en-us/library/windows/desktop/aa366887(v=vs.85).aspx
-        ctypes.windll.kernel32.VirtualAlloc.argtypes = [
-            ctypes.c_void_p, ctypes.c_long, ctypes.c_long, ctypes.c_long]
-        ctypes.windll.kernel32.VirtualAlloc.restype = ctypes.c_void_p
-        self.addr = ctypes.windll.kernel32.VirtualAlloc(
-            0, ctypes.c_long(self.size_bytes), mem_commit, page_readwrite)
+        # https://stackoverflow.com/questions/61590363/enforce-virtualalloc-address-less-than-32-bits-on-64-bit-machine
+        VirtualAlloc = ctypes.windll.kernel32.VirtualAlloc
+        VirtualAlloc.argtypes = [wt.LPVOID, ctypes.c_size_t, wt.DWORD, wt.DWORD]
+        VirtualAlloc.restype = wt.LPVOID
+
+        self.addr = VirtualAlloc(
+            0,
+            ctypes.c_size_t(self.size_bytes), 
+            mem_commit, 
+            page_readwrite
+        )
+
+        # Log buffer information
+        if self.logger:
+            address_bits = None if self.addr is None else round(self.addr/2**32, 3)
```

Comment:
I don't understand this variable why are you using a floating point type?

Reply:
In this case I wanted to find out if it exceeded 32 bits, or otherwise how close it was to 32 bits. That's why I didn't want to doa a simple bit shift